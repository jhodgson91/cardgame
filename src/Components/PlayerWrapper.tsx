import React, { Component } from 'react';
import './Main.scss'

//Define props types
export interface Props {
  title: string;
  grid: number;
}

export default class PlayerWrapper extends React.Component<Props> {

  constructor(props: Props) {
    super(props);
  }

  render(): React.ReactNode {
    // This wants to use the props.children thing we found to render its contents
    return (
      <section id="players">
        <div className="grid-container">
          <div className="grid-x grid-padding-x align-center">
            <div className={`cell small-${this.props.grid}`}>
              <h2>{this.props.title}</h2>
              {this.props.children}
            </div>
          </div>
        </div>
      </section>
    );
  }
}
