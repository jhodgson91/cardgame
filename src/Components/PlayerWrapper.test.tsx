import * as React from 'react';
import * as enzyme from 'enzyme';
import PlayerWrapper from './PlayerWrapper';

//TODO: Test injecting cards with keys

it('renders with a title and a player mock', () => {
  const playerWrapper = enzyme.shallow(<PlayerWrapper title='house' players={<div id='test'>Test</div>}/>);
  expect(playerWrapper.find('#players')).toBeDefined;
  expect(playerWrapper.find('h2').text()).toBe('house');
  expect(playerWrapper.find('div#test')).toBeDefined;
  expect(playerWrapper.find('#test').text()).toBe('Test');
});
