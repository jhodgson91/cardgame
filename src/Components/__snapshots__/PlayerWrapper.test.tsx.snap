// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the same as last time 1`] = `
<section
  id="players"
>
  <div
    className="grid-container"
  >
    <div
      className="grid-x grid-padding-x align-center"
    >
      <div
        className="cell small-12"
      >
        <h2>
          house
        </h2>
        <div
          id="test"
        >
          Text
        </div>
      </div>
    </div>
  </div>
</section>
`;
